src/com/rcv/ElectionConfig.java

```diff
@@ -408,24 +424,27 @@ String getNameForCandidateID(String candidateID) {
   // purpose: builds map of candidate ID to candidate name and possibly generates tie-break ordering
   private void processCandidateData() {
     candidateCodeToNameMap = new HashMap<>();
-    // candidate is used to index through all candidates for this election
-    for (RawElectionConfig.Candidate candidate : rawConfig.candidates) {
-      if (candidate.code != null) {
-        candidateCodeToNameMap.put(candidate.code, candidate.name);
-        candidatePermutation.add(candidate.code);
-      } else {
-        candidateCodeToNameMap.put(candidate.name, candidate.name);
-        candidatePermutation.add(candidate.name);
+
+    if (rawConfig.candidates != null) {
```

Comment:
we validate that candidates may not be null, so I prefer to remove this check or replace with an assertion

Reply:
If you leave the "candidates" field out of the config file entirely, rawConfig.candidates will be null -- so we do need to check for this. Or did I misunderstand your comment?

Reply:
Damn, I forgot that was an option.  In that case never mind@